libp2p/protocols/kademlia.nim

```diff
@@ -79,47 +92,49 @@ proc new*(
   let kad = T(
     rng: rng,
     switch: switch,
+    bootstrapNodes: bootstrapNodes,
     rtable: rtable,
     config: config,
     providerManager:
       ProviderManager.new(config.providerRecordCapacity, config.providedKeyCapacity),
   )
 
-  kad.codec = KadCodec
-  kad.handler = proc(
-      conn: Connection, proto: string
-  ) {.async: (raises: [CancelledError]).} =
-    defer:
-      await conn.close()
-    while not conn.atEof:
-      let buf =
-        try:
-          await conn.readLp(MaxMsgSize)
-        except LPStreamEOFError:
+  kad.codec = codec
+  if not client:
```

Comment:
Couldnt we do something like:
```
if client:
  return kad
```
to avoid all of this code living inside the `if`?